Index: source/players/iframe.js
===================================================================
--- source/players/iframe.js	(revision 1134)
+++ source/players/iframe.js	(working copy)
@@ -41,7 +41,7 @@
 
         var html = '<iframe id="' + this.id + '" name="' + this.id + '" height="100%" ' +
             'width="100%" frameborder="0" marginwidth="0" marginheight="0" ' +
-            'style="visibility:hidden;" ' +
+            'style="visibility:visible;" ' +
 			'onload="this.style.visibility=\'visible\'" ' +
             'scrolling="' + scrolling + '"';
 
